src/graphql/data/nft/Asset.ts

```diff
@@ -116,6 +123,61 @@ type NftAssetsQueryAsset = NonNullable<
   NonNullable<NonNullable<AssetQuery_nftAssets$data['nftAssets']>['edges']>[number]
 >
 
+function formatAssetQueryData(queryAsset: NftAssetsQueryAsset, totalCount?: number) {
+  const asset = queryAsset.node
+  const ethPrice = parseEther(
+    asset.listings?.edges[0]?.node.price.value?.toLocaleString('fullwide', { useGrouping: false }) ?? '0'
+  ).toString()
+  return {
+    id: asset.id,
+    address: asset?.collection?.nftContracts?.[0]?.address,
+    notForSale: asset.listings?.edges?.length === 0,
+    collectionName: asset.collection?.name,
+    collectionSymbol: asset.collection?.image?.url,
+    imageUrl: asset.image?.url,
+    animationUrl: asset.animationUrl,
+    marketplace: asset.listings?.edges[0]?.node?.marketplace?.toLowerCase(),
+    name: asset.name,
+    priceInfo: asset.listings
+      ? {
+          ETHPrice: ethPrice,
+          baseAsset: 'ETH',
+          baseDecimals: '18',
+          basePrice: ethPrice,
+        }
+      : undefined,
+    susFlag: asset.suspiciousFlag,
+    sellorders: asset.listings?.edges.map((listingNode) => {
+      return {
+        ...listingNode.node,
+        protocolParameters: listingNode.node?.protocolParameters
+          ? JSON.parse(listingNode.node?.protocolParameters.toString())
+          : undefined,
+      }
+    }),
+    smallImageUrl: asset.smallImage?.url,
+    tokenId: asset.tokenId,
+    tokenType: asset.collection?.nftContracts?.[0]?.standard,
+    totalCount,
+    collectionIsVerified: asset.collection?.isVerified,
+    rarity: {
+      primaryProvider: 'Rarity Sniper', // TODO update when backend adds more providers
+      providers: asset.rarities?.map((rarity) => {
+        return {
+          ...rarity,
+          provider: 'Rarity Sniper',
+        }
+      }),
+    },
+    owner: asset.ownerAddress,
+    creator: {
+      profile_img_url: asset.collection?.creator?.profileImage?.url,
+      address: asset.collection?.creator?.address,
+    },
+    metadataUrl: asset.metadataUrl,
+  }
+}
+
 export function useAssetsQuery(
```

Comment:
did this change other than moving it? hard to tell

Reply:
it just moved the formatting logic to its own helper function